<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>751</width>
    <height>685</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SYSID</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="leftMargin">
     <number>16</number>
    </property>
    <property name="topMargin">
     <number>16</number>
    </property>
    <property name="rightMargin">
     <number>16</number>
    </property>
    <property name="bottomMargin">
     <number>16</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontal_layout">
        <item>
         <layout class="QVBoxLayout" name="left_layout">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
          </property>
          <item>
           <layout class="QHBoxLayout" name="correctors_header_layout">
            <item>
             <widget class="QLabel" name="correctors_label">
              <property name="text">
               <string>Correctors</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QListWidget" name="correctors_list">
            <property name="selectionMode">
             <enum>QAbstractItemView::SelectionMode::ExtendedSelection</enum>
            </property>
           </widget>
          </item>

          <item>
           <layout class="QHBoxLayout" name="correctors_buttons_layout">
            <item>
             <widget class="QPushButton" name="save_correctors_button">
              <property name="text">
               <string>Save As..</string>
              </property>
             </widget>
            </item>

            <item>
             <widget class="QPushButton" name="load_correctors_button">
              <property name="text">
               <string>Load..</string>
              </property>
             </widget>
            </item>

            <item>
             <widget class="QPushButton" name="clear_correctors_button">
              <property name="text">
               <string>Clear</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>


          <item>
           <layout class="QHBoxLayout" name="bpms_header_layout">
            <item>
             <widget class="QLabel" name="bpms_label">
              <property name="text">
               <string>BPMs</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QListWidget" name="bpms_list">
            <property name="selectionMode">
             <enum>QAbstractItemView::SelectionMode::ExtendedSelection</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="bpms_buttons_layout">
            <item>
             <widget class="QPushButton" name="save_bpms_button">
              <property name="text">
               <string>Save As..</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="load_bpms_button">
              <property name="text">
               <string>Load..</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="clear_bpms_button">
              <property name="text">
               <string>Clear</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>

        <!--SAVING THE DATA -->
        <item>
         <layout class="QVBoxLayout" name="right_layout">
          <property name="spacing">
           <number>2</number>
          </property>
          <item>
           <widget class="QLabel" name="info_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Data Storage:</string>
            </property>
           </widget>
          </item>
          <!-- END OF IT-->
          <item>
           <layout class="QHBoxLayout" name="working_directory_layout">
            <item>
             <widget class="QLineEdit" name="working_directory_input">
              <property name="placeholderText">
               <string>Working directory:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="working_directory_dialog">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="options_layout">
            <item>
             <widget class="QLabel" name="options_label">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>


              <property name="text">
               <string>Options</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="cycle_mode_layout">
              <item>
               <widget class="QLabel" name="cycle_mode_label">
                <property name="text">
                 <string>Cycle mode</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="cycle_mode_combobox">
                <item>
                 <property name="text">
                  <string>Repeat all</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Repeat selected</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="current_layout">
              <item>
               <widget class="QLabel" name="current_label">
                <property name="text">
                 <string>Max strength (gauss*m)</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="current_spacer">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                 <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="horizontal_current_label">
                <property name="text">
                 <string>H:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="max_horizontal_current_spinbox">
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="vertical_current_label">
                <property name="text">
                 <string> V:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="max_vertical_current_spinbox">
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="excursion_layout">
              <item>
               <widget class="QLabel" name="excursion_label">
                <property name="text">
                 <string>Target orbit excursion (mm)</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="excursion_spacer">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="horizontal_excursion_label">
                <property name="text">
                 <string>H:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="horizontal_excursion_spinbox">
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
                <property name="value">
                 <double>0.500000000000000</double>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="vertical_excursion_label">
                <property name="text">
                 <string>V:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="vertical_excursion_spinbox">
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
                <property name="value">
                 <double>0.500000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QWidget" name="plot_widget" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>400</width>
                <height>300</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="buttons_layout">
        <item>
         <widget class="QPushButton" name="start_button">
          <property name="styleSheet">
           <string notr="true">background-color: red; color: white;</string>
          </property>
          <property name="text">
           <string>START</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="stop_button">
          <property name="styleSheet">
           <string notr="true">background-color: green; color: white;</string>
          </property>
          <property name="text">
           <string>STOP</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
